Plugin Management: Minor UI improvements and fixes.

jkguidaven opened the pull request:
This PR fixes a couple of UI issues related to the Plugin management components. 

Closes https://linear.app/a8c/issue/A4A-777/plugin-management-panel-view-ui-problems

## Proposed Changes

| Issue | Before | After |
|--------|--------|--------|
| If the Plugin name is excessively long, it may overlap when on list view, resulting in a horizontal scrollbar. | <img width="420" alt="Screenshot 2025-04-21 at 10 28 44 PM" src="https://github.com/user-attachments/assets/c1abd244-9afd-4b38-af4f-5e0e435a5854" /> | <img width="416" alt="Screenshot 2025-04-21 at 10 29 13 PM" src="https://github.com/user-attachments/assets/9cecd614-b174-400a-b565-c6c3a8a6faea" /> |
| Update CTAs near each plugin in sidebar should removed. The information density of this area is too much, and I'd be willing to bet that most folks aren't going to mass update plugins across all sites from this area. It's not visually aligned or the right size. They'll want to know details and update, which they can do once they click on the plugin title. It doesn't represent how many instances need that update. IE: It says Woo is installed on "11 sites "update to version 9.8.1" but actually only one of those sites needs an update. Let's leave the update action to the right panel. See screenshot. | <img width="410" alt="Screenshot 2025-04-21 at 10 32 32 PM" src="https://github.com/user-attachments/assets/0ced9f9d-7d26-456a-aeb9-19e318e0ff3c" /> | <img width="411" alt="Screenshot 2025-04-21 at 10 32 39 PM" src="https://github.com/user-attachments/assets/8f585a90-6946-49a9-a8a5-680fc0d58d41" /> |
| **A4A only:** The double border on the pagination shows here as well. | <img width="280" alt="Screenshot 2025-04-21 at 10 35 13 PM" src="https://github.com/user-attachments/assets/788a9766-a23a-4646-87e2-c8c251545a70" /> | <img width="232" alt="Screenshot 2025-04-21 at 10 35 21 PM" src="https://github.com/user-attachments/assets/2bd3ab79-691d-46cc-a10e-fe3203b7598c" /> | 

## Why are these changes being made?

* This enhances our UI, providing a more refined and enjoyable user experience.

## Testing Instructions

* Access the A4A live link and proceed to `/plugins`.
* Verify all elaborated issues are fixed.

## Pre-merge Checklist


- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

jkguidaven committed:
Make sure plugin name do not overflow.

jkguidaven committed:
Remove 'Pending update' button.

jkguidaven committed:
Hide Update CTAs when on list view.

jkguidaven committed:
Fix double border on Dataview pagination.

paulopmt1 commented:
Thanks for adding those improvements @jkguidaven!

> The "Pending update" button should be removed from the sidebar. It's redundant to the filtering and occupies too much space in this area.

This CTA easily allows users to see one of the most important features on this screen: "Which plugins do we have update for?". Although we have that feature in the filter list, I recall that during our usability tests, some users didn't see the filter at all, which is why we kept the redundancy button. If it's using too much space, perhaps we could shorten it or completely hide it on mobile devices? cc @jeffgolenski

jeffgolenski commented:
@paulopmt1 

> This CTA easily allows users to see one of the most important features on this screen: "Which plugins do we have update for?". Although we have that feature in the filter list, I recall that during our usability tests, some users didn't see the filter at all, which is why we kept the redundancy button. If it's using too much space, perhaps we could shorten it or completely hide it on mobile devices? cc @jeffgolenski

Totally. I know the filtering isn't ideal and can understand the reason for that—It's a bandage to a larger issue. To be clear, I wasn't suggesting we remove it completely, I'm just concerned about how much space it occupies when the middle pane is visible—it takes away from Search and also with translations to languages other than english it'll be crazy.

Quick video of what I mean!


https://github.com/user-attachments/assets/87ac819d-eb38-442b-b8e5-8cc627dc4c39



I think we can leave it for now @jkguidaven, but I'd really like to build some tracks data into that action so we can see how much use it's getting. It is a redundant action, so it's still concerning for me.

yashwin commented on client/my-sites/plugins/plugins-list/use-fields.tsx:
```diff
@@ -121,7 +121,11 @@ export function useFields(
 				enableHiding: false,
 				render: ( { item }: { item: Plugin } ) => {
-					if ( item.status?.includes( PLUGINS_STATUS.UPDATE ) && item?.update?.new_version ) {
+					if (
+						item.status?.includes( PLUGINS_STATUS.UPDATE ) &&
+						item?.update?.new_version &&
+						! isListView
```
We want to always hide this, right?

jkguidaven replied:
The button is only hidden in list view when we have limited space.

yashwin reviewed (changes requested):
Thanks for the improvements, @jkguidaven!

The pagination on the mobile is hidden. Can we also fix this? It was slightly visible earlier, but it's completely gone:

<img width="420" alt="Screenshot 2025-04-25 at 10 04 48 AM" src="https://github.com/user-attachments/assets/6df9272e-667b-4915-81e4-d101f09c3fb5" />

yashwin commented on client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx:
```diff
@@ -107,58 +100,12 @@ export default function PluginsListDataViews( {
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [ pluginSlug ] );
 
-	const header = (
-		<>
```
We could probably hide this when the preview is open or the filter is applied instead of removing this.

jkguidaven replied:
I've reverted the changes, as discussed above.

jkguidaven committed:
Revert change on 'Update' CTA button.

jkguidaven committed:
Add tracking for clicking the 'Pending update' button.

jkguidaven commented:
> @paulopmt1
> 
> > This CTA easily allows users to see one of the most important features on this screen: "Which plugins do we have update for?". Although we have that feature in the filter list, I recall that during our usability tests, some users didn't see the filter at all, which is why we kept the redundancy button. If it's using too much space, perhaps we could shorten it or completely hide it on mobile devices? cc @jeffgolenski
> 
> Totally. I know the filtering isn't ideal and can understand the reason for that—It's a bandage to a larger issue. To be clear, I wasn't suggesting we remove it completely, I'm just concerned about how much space it occupies when the middle pane is visible—it takes away from Search and also with translations to languages other than english it'll be crazy.
> 
> Quick video of what I mean!
> 
>  pending-update-button.mp4 
> I think we can leave it for now @jkguidaven, but I'd really like to build some tracks data into that action so we can see how much use it's getting. It is a redundant action, so it's still concerning for me.

Thanks. I have reverted my change and also added the `calypso_plugins_list_pending_update_filter_click` track event so we can begin tracking this button.

jkguidaven committed:
Add missing dispatch.

jkguidaven committed:
Fix dataview padding bottom in mobile.

jkguidaven commented:
> Thanks for the improvements, @jkguidaven!
> 
> The pagination on the mobile is hidden. Can we also fix this? It was slightly visible earlier, but it's completely gone:
> 
> <img alt="Screenshot 2025-04-25 at 10 04 48 AM" width="420" src="https://private-user-images.githubusercontent.com/10586875/437300314-6df9272e-667b-4915-81e4-d101f09c3fb5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU1Nzg0MTksIm5iZiI6MTc0NTU3ODExOSwicGF0aCI6Ii8xMDU4Njg3NS80MzczMDAzMTQtNmRmOTI3MmUtNjY3Yi00OTE1LTgxZTQtZDEwMWYwOWMzZmI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDEwNDgzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViNDU2NThkZjQ1MWZhNGM2NGU0ZmIxMjExMWNiOTNlNDdjOTUyZDY5MjM5MDY5Y2E3ZjBiOGFjOGI5YjU4YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.62KokJA_VlxwjtC8gOocH1Hx7d52PZRvbMaZjyu8a7w">

Thanks, I've resolved the issue.

yashwin reviewed (approved):
Thanks for addressing my comments. LGTM! ✅

jeffgolenski reviewed (changes requested):
Thanks @jkguidaven - Mostly looks great, I just noticed two small things.

* The pagination in the middle pane has a gap at the bottom. Looks like we just need the plugin list container to fill that space. 
* The weird behavior of the rounded corners is showing up at the bottom of this table as well.

![CleanShot 2025-04-28 at 13 00 57@2x](https://github.com/user-attachments/assets/37084862-10ae-461a-87aa-2fd529e57cc8)


Otherwise all good!

jkguidaven committed:
Fix rebase issue.

jkguidaven committed:
Fix bottom spacing when on preview mode.

jkguidaven commented:
> Thanks @jkguidaven - Mostly looks great, I just noticed two small things.
> 
> * The pagination in the middle pane has a gap at the bottom. Looks like we just need the plugin list container to fill that space.
> * The weird behavior of the rounded corners is showing up at the bottom of this table as well.
> 
> ![CleanShot 2025-04-28 at 13 00 57@2x](https://private-user-images.githubusercontent.com/214813/438316282-37084862-10ae-461a-87aa-2fd529e57cc8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYwMjU5NzksIm5iZiI6MTc0NjAyNTY3OSwicGF0aCI6Ii8yMTQ4MTMvNDM4MzE2MjgyLTM3MDg0ODYyLTEwYWUtNDYxYS04N2FhLTJmZDUyOWU1N2NjOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQzMFQxNTA3NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYjk4ZjVmNDliMjFmYTI5NzY5ZDYzY2VmMzg3MWI1MTNjYTU1MThmZmQwZjc0ZGU1NmRhN2FjMWI3MDEzY2E1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WcABzYJvuxRG4RGkG8gF2aua73XCsDDUSHMtHw6cND4)
> 
> Otherwise all good!

Thanks for the review. Both issues have been addressed 👍
